crates/bifrost/src/providers/replicated_loglet/sequencer/appender.rs

```diff
@@ -213,18 +251,16 @@ impl<T: TransportConnect> SequencerAppender<T> {
         }
 
         // otherwise, we try to send the wave.
-        self.send_wave(servers).await
+        self.send_wave(checker, servers).await
     }
 
-    async fn send_wave(&mut self, spread_servers: Vec<RemoteLogServer>) -> SequencerAppenderState {
+    async fn send_wave(
+        &mut self,
+        checker: &mut NodeSetChecker<'_, NodeAttributes>,
```

Comment:
The checker will be created based of a certain view of the NodesConfig. This may change across waves, so we might want to create the construct the checker on every wave.